modules/library/render/src/main/java/org/geotools/renderer/lite/StreamingRenderer.java

```diff
@@ -1067,10 +1088,46 @@ Query getStyleQuery(
             List<ReferencedEnvelope> envelopes = null;
             // enable advanced projection handling with the updated map extent
             if (isAdvancedProjectionHandlingEnabled()) {
+                Map projectionHints = new HashMap();
+                if (isAdvancedProjectionDensificationEnabled()
+                        && !CRS.equalsIgnoreMetadata(featCrs, mapCRS)) {
+                    ReferencedEnvelope targetEnvelope = envelope;
+                    ReferencedEnvelope sourceEnvelope = targetEnvelope.transform(featCrs, true);
+                    AffineTransform at = worldToScreenTransform;
+                    MathTransform2D crsTransform =
+                            (MathTransform2D)
+                                    CRS.findMathTransform(
+                                            CRS.getHorizontalCRS(featCrs),
+                                            CRS.getHorizontalCRS(mapCRS));
+                    MathTransform2D screenTransform = new AffineTransform2D(at);
+                    MathTransform2D fullTranform =
+                            (MathTransform2D)
+                                    ConcatenatedTransform.create(crsTransform, screenTransform);
+                    Rectangle2D.Double sourceDomain =
+                            new Rectangle2D.Double(
+                                    sourceEnvelope.getMinX(),
+                                    sourceEnvelope.getMinY(),
+                                    sourceEnvelope.getWidth(),
+                                    sourceEnvelope.getHeight());
+                    WarpBuilder wb = new WarpBuilder(0.8);
+                    int[] actualSplit = wb.getRowColsSplit(fullTranform, sourceDomain);
+                    if (actualSplit != null && (actualSplit[0] != 1 || actualSplit[1] != 1)) {
```

Comment:
For example, I've tried a UTM, zoomed out enough that I could see straight lines appearing because warp builder gave up:

![image](https://user-images.githubusercontent.com/325433/53572231-17e60200-3b6b-11e9-9043-ef297999069c.png)

and here is the result forcing an arbitrary densification distance of 5 in case null was returned:

![image](https://user-images.githubusercontent.com/325433/53572275-35b36700-3b6b-11e9-8879-d4ba4a32f02a.png)

I've played a bit with different values, a fixed distance of 10 still produced visible "corners" or "turning points", while the difference between 5 and 2 was hard to notice. I'm guessing 5 is a good value.